docs/requirements.md

```diff
@@ -0,0 +1,61 @@
+At its heart ServiceControl and ServicePulse provide operational support for systems built on the Particular Service Platform. They do this by:
+
+- monitoring system health
+- diagnosing and treating errors
+
+In this context, system refers to NServiceBus endpoints and communication with third-party services.
+
+ServiceControl should provide the following features:
+
+- process and store the contents of the error queue
+- replay messages from the error queue
+- process and store the contents of the audit queue
+- monitor the availability of endpoints
+- generate and collect saga state changes
+- provide an API for monitoring user-defined conditions
+- provide integration hooks for developers to consume
+- search for messages
+- export failed message data for external analysis
+- report on the contents of the timeout manager
+- allow user to provide metadata about messages for operational support
+- allow user to easily provide feedback and usage data and diagnostics for the platform
+- monitor the health of SC itself
+
+Implicit in these requirements is that the Service Platform should:
+
+- scale to an entire system based on the Particular Service Platform
+- be durable (i.e. no message loss)
+- be secure (i.e. calls to ServiceControl should be authenticated)
+- provide a smooth installation and upgrade experience
+- be performant
+- provide a clear and intuitive UI
+- be responsive with UI components (not in the HTML/mobile sense); i.e. it should always indicate when it is "doing something"
+- support all versions of all transports and all transport features
+
+### Process and store the contents of the error queue
+
+Rationale: To view the contents of the error queue in order to triage potential bugs or infrastructure failures in a system.
+
+### Replay messages from the error queue
+
+Rationale: To recover from bugs or infrastructure failures in order to maintain stability of a system.
+
+### Process and store the contents of the audit queue
+
+Rationale: To verify that a replayed message completed successfully. To understand the emergent design of a system through correllated messages. To identify design flaws in a system.
+
+### Generate and collect heartbeat messages
+
+Rationale: Verify that endpoints are up and able to send messages. Facilitates notification in the event an endpoint is no longer operational.
+
+### Generate and collect saga state changes
+
+Rationale: To understand the operational behavior of a long-running process and to troubleshoot design flaws.
+
+### Provide an API for monitoring user-defined conditions
+
+Rationale: To monitor endpoint dependencies. Allows team to more quickly identify and react to infrastructure failures.
+
+### Provide operational events for consumption by other applications
+
+Rationale: To provide extensibility for developers to react to events in ServiceControl. Allows users to provide better diagnostics for their systems.
```

Comment:
I really want to explore this further. I hate the current set of Failed Message events and would rather see them based on Groups. I also think a well defined and public API for SC would help us to reduce the number of events published here.